### DISCUSSION TITLE
Build Error suddenly on Vercel

### DISCUSSION THREAD

**Discussion by Quilljou** [vercel/community#4356] [Help]
### Summary

The same commit can build some days ago, but can not today, Have no idea what happened, It can build in my own MacOS machine
 anyone help? Thanks
![image](https://github.com/vercel/community/assets/18414813/595b4b58-45cd-4562-9653-2e66e4267d1a)

### Steps to Reproduce

Reproduce in my account

**Comment by amyegan** [comment]
It could be that a dependency is missing in the `package.json` file. Please check that all necessary packages are included 🙏

**Reply by Quilljou** [reply]
Thanks for replying, After remove node_modules in my Mac, I did reproduce the error. But seems no dependency is missing, here is the missing `ie` function
<img width="1034" alt="Screenshot 2023-10-04 at 22 10 00" src="https://github.com/vercel/community/assets/18414813/35c248e1-45f3-4324-8ac4-1bcf2364d1cd">
<img width="859" alt="Screenshot 2023-10-04 at 22 10 13" src="https://github.com/vercel/community/assets/18414813/ccc3220e-ea16-4d14-8ce7-00e26b0a3f77">

**Reply by amyegan** [reply]
Do you have a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that you can share?

**Reply by Quilljou** [reply]
I have tried to reproduce with an empty next project, But I can't reproduce it. What I found is I'm using  react-dropzone, In my broken project I find nextjs build with react-dropzone umd module, But the new good empty project is with react-dropzone es module. any clue? @amyegan

**Reply by Quilljou** [reply]
@amyegan  Hi, here is the reproduce error repo: https://github.com/Quilljou/reproduce-build-error

**Reply by amyegan** [reply]
Since you were able to reproduce the issue in your local development environment, that does sound like a dependency problem. It may have been caused by a `minor` or `patch` update to one of the required packages. I tried removing the `^` from the beginning of all `dependencies` and `devDependencies` and that allowed me to deploy the example without getting the error.

Since there are still quite a lot of dependencies in the example, I'm not sure which one caused the problem. But knowing that it was caused by a new package version should help you locate the source of the error. 

More information about npm semantic versioning can be found at https://docs.npmjs.com/about-semantic-versioning

**Reply by Quilljou** [reply]
semver is a bad design honestly

**Reply by Quilljou** [reply]
But we should figure out why, is isn't.
